### ISSUE TITLE
[camera] Use background method channel on iOS

### ISSUE THREAD

**Issue by cyanglaz** [flutter/flutter#95424]
Camera plugin maintains a dispatch_queue to manually dispatched all the method channel calls to the background thread: 
https://github.com/flutter/plugins/blob/master/packages/camera/camera/ios/Classes/CameraPlugin.m#L1354-L1366
This introduce additional thread hops. It is also error-prone. See: https://github.com/flutter/flutter/issues/59124 and description in https://github.com/flutter/plugins/pull/4619

We should migrate this plugin to use the background method channel introduced in https://github.com/flutter/engine/pull/29665, which was reverted in https://github.com/flutter/engine/pull/30289

Once a similar solution lands in engine, the camera plugin should utilize it so it doesn't need to maintain its own queue. 

CC @gaaclarke @hellohuanlin

**Comment by stuartmorgan** [comment]
Dropping to P5, since the plan to fix the threading model turned out not to benefit much from background channels. It's still an nice to have to reduce thread hopping slightly though.

**Comment by gaaclarke** [comment]
> turned out not to benefit much from background channels

Just an asterisk.  The results were complex.  On iOS local testing microbenchmarks the serial channels were 40% faster, but on devicelab they were the same.  On android local testing the microbenchmarks for serial channels was slower ( can't recall the percentage ), but on devicelab they were faster.  Also on Android the results varied based on the OS throttling of the platform thread.

That's if you are using serial channels, if you use concurrent channels the performance gains are more clear.

There is also the macro perspective where many plugins using background platform channels will perform better since they are using a shared thread pool.  So, they are better citizens in that respect.

How should users interpret that?  They should use them when authoring new code.  When migrating old code, it is higher priority if you can take advantage of concurrent task queues, but everyone that can use them should eventually.

**Comment by stuartmorgan** [comment]
> > turned out not to benefit much from background channels
> 
> Just an asterisk. The results were complex.

This comment wasn't about performance, it was about the iOS implementation design overall (specifically "the camera plugin should utilize it so it doesn't need to maintain its own queue"). Camera turns out to need several queues for specific purposes, and because of their needs the design doesn't fundamentally change much when using background platform channels.
